### PR TITLE
Publish java agent to maven central

### PR DESCRIPTION
**Pull request by ravisingal**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -107,7 +107,9 @@ workflows:
             tags:
               only: /.*/
       - release-bintray-publish:
-          context: hypertrace-publishing
+          context:
+            - hypertrace-publishing
```
Comment (aaron-steinfeld) [inline .circleci/config.yml:111]: do we still need the old context? Also, are there plans on moving this repo off circleci? i think it's the last one preventing us from completely disconnecting cci.

Comment (ravisingal) [reply .circleci/config.yml:113]: This context has `DOCKER_USERNAME` and `DOCKER_PASSWORD`. Not sure if we need `docker login` here.

will move this repo to Github Actions as part of another PR (maybe next week).

File: .circleci/config.yml [hunk 2]
```diff
@@ -99,15 +99,19 @@ workflows:
   build-and-publish:
     jobs:
       - build:
+          context: maven-central-publish
```
Comment (aaron-steinfeld) [inline .circleci/config.yml:102]: we should only add this on the release job. If the plugin requires this when publish isn't invoked, we should update that on the plugin side (as users will not have these values set)

File: .circleci/config.yml [hunk 3]
```diff
@@ -99,15 +99,19 @@ workflows:
...
           filters:
             tags:
               only: /.*/
       - muzzle:
+          context: maven-central-publish
           filters:
             tags:
               only: /.*/
       - release-bintray-publish:
```
Comment (aaron-steinfeld) [inline .circleci/config.yml:111]: this job is no longer going to bintray so should be renamed

File: .circleci/config.yml (hunk 1 above)
Comment (aaron-steinfeld) [reply .circleci/config.yml:113]: I see it's setting up and logging into docker, but I can't imagine why. Guessing that was just copy pasted from a job building docker images - this one isn't, right?

Comment (ravisingal) [reply .circleci/config.yml:113]: This repo also builds docker image: https://github.com/hypertrace/javaagent/blob/main/Makefile#L22-L31

It is used for copying java agent to container filesystem before starting the java process using a mutating webhook.

File: .circleci/config.yml (hunk 3 above)
Comment (ravisingal) [reply .circleci/config.yml:111]: done.

File: .circleci/config.yml (hunk 1 above)
Comment (aaron-steinfeld) [reply .circleci/config.yml:113]: I dare not ask why we're using make here..

File: .circleci/config.yml (hunk 2 above)
Comment (ravisingal) [reply .circleci/config.yml:102]: Raised the following PR: https://github.com/hypertrace/hypertrace-gradle-publish-plugin/pull/16

**Review by aaron-steinfeld — APPROVED** [review]

**Pull request merged**